pkg/build/registry/buildconfig/webhook.go

```diff
@@ -0,0 +1,75 @@
+package buildconfig
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic/rest"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/build/client"
+	"github.com/openshift/origin/pkg/build/webhook"
+)
+
+func NewWebHookREST(registry Registry, instantiator client.BuildConfigInstantiator, plugins map[string]webhook.Plugin) *rest.WebHook {
+	controller := &controller{
+		registry:     registry,
+		instantiator: instantiator,
+		plugins:      plugins,
+	}
+	return rest.NewWebHook(controller, false)
+}
+
+type controller struct {
+	registry     Registry
+	instantiator client.BuildConfigInstantiator
+	plugins      map[string]webhook.Plugin
+}
+
+// ServeHTTP implements rest.HookHandler
+func (c *controller) ServeHTTP(w http.ResponseWriter, req *http.Request, ctx kapi.Context, name, subpath string) error {
+	parts := strings.Split(subpath, "/")
+	if len(parts) < 2 {
+		return errors.NewBadRequest(fmt.Sprintf("unexpected hook subpath %s", subpath))
+	}
+	secret, hookType := parts[0], parts[1]
+	path := ""
+	if len(parts) > 2 {
```

Comment:
Was left over from the old thing.  I believe we had that originally so that in case we implemented "MagicWebhook" that needed to pass things on the path.

----- Original Message -----
> > +
> > +type controller struct {
> > +	registry     Registry
> > +	instantiator client.BuildConfigInstantiator
> > +	plugins      map[string]webhook.Plugin
> > +}
> > +
> > +// ServeHTTP implements rest.HookHandler
> > +func (c *controller) ServeHTTP(w http.ResponseWriter, req *http.Request,
> > ctx kapi.Context, name, subpath string) error {
> > +	parts := strings.Split(subpath, "/")
> > +	if len(parts) < 2 {
> > +		return errors.NewBadRequest(fmt.Sprintf("unexpected hook subpath %s",
> > subpath))
> > +	}
> > +	secret, hookType := parts[0], parts[1]
> > +	path := ""
> > +	if len(parts) > 2 {
> 
> what do we expect these additional path segments to be/represent?
> 
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2082/files#r29800666